File: misc/balanced_brackets/balanced_brackets.js
const allowed_next = new Map([
    ["(", ")"],
    ["{", "}"],
    ["[","]"]
]);

/**
 * Checks if a series of brackets are open and closed in the right order and balanced.
 * 
 * @param {string} brackets 
 * @return {bool} true in case the brackets are right.
 * 
 * > validBrackets("{{[[(())]]}}")
 * true
 * > validBrackets("{[(])}")
 * false
 * > validBrackets("()")
 * true
 * > validBrackets("[({})](")
 * false
 */
function validBrackets(brackets) {
    const pending = [];
    for(let i=0; i<brackets.length; i++) {
        const current = brackets[i];
        
        if (allowed_next.has(current)) {
            pending.push(allowed_next.get(current));
            continue;
        }
        
        const expected = pending.pop();
        if (expected !== current) {
            return false;
        }
        
    }

    return pending.length == 0;
}

exports = validBrackets;

File: misc/balanced_brackets/balanced_brackets.exs
defmodule BalancedBrackets do
  @doc """
  Checks if a series of brackets are correctly balanced
  """
  def isBalanced(input) when is_binary(input), do:
    isBalanced(String.graphemes(input), [])
  
  def isBalanced(["(" | rest], pending), do:
    isBalanced(rest, [")" | pending])
  
  def isBalanced(["{" | rest], pending), do:
    isBalanced(rest, ["}" | pending])
  
  def isBalanced(["[" | rest], pending), do:
    isBalanced(rest, ["]" | pending])
  
  def isBalanced(["}" | rest], ["}" | pending]), do:
    isBalanced(rest, pending)
  
  def isBalanced([")" | rest], [")" | pending]), do:
    isBalanced(rest, pending)
  
  def isBalanced(["]" | rest], ["]" | pending]), do:
    isBalanced(rest, pending)
  
  def isBalanced([], []), do: true
  def isBalanced(_, _), do: false
end

ExUnit.start()

defmodule BalancedBracketsTest do
  use ExUnit.Case, async: true

  test "{{[[(())]]}}", do:
    assert BalancedBrackets.isBalanced("{{[[(())]]}}")
  test "{[(])}", do:
    refute BalancedBrackets.isBalanced("{[(])}")
  test "()", do:
    assert BalancedBrackets.isBalanced("()")
  test "[({})](", do:
    refute BalancedBrackets.isBalanced("[({})](")
end